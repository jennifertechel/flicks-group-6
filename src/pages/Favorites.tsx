import { Flex, Grid, Heading } from "@chakra-ui/react";
import data from "../../data/movies.json";
import Thumbnail from "../components/Thumbnail";
import { useLocalStorage } from "../hooks/useLocalStorage";
function Favorites() {
  const [likedMovies] = useLocalStorage("likedMovies", []);

  const favoriteMovies = data.filter((movie) =>
    likedMovies.includes(movie.title)
  );

  return (
    <Flex flexDir="column" px={35}>
      <Heading pb={4}>Favorites</Heading>
      <Flex flexDir="column">
        <Grid templateColumns="repeat(auto-fill, minmax(200px, 1fr))" gap={4}>
          {favoriteMovies.map((movie, index) => (
            <Thumbnail
              key={index}
              image={movie.thumbnail}
              rating={movie.rating}
              year={movie.year}
              title={movie.title}
              genre={movie.genre}
            />
          ))}
        </Grid>
      </Flex>
    </Flex>
  );
}

export default Favorites;
