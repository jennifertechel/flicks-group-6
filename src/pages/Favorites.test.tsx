import { render } from "@testing-library/react";
import Favorites from "./Favorites";

describe("Favorites component", () => {
  it("renders correctly with Thumbnail components", () => {
    const { container, rerender } = render(<Favorites />);

    expect(container).toMatchSnapshot();

    rerender(<Favorites />);

    const thumbnails = container.querySelectorAll(".thumbnail");
    if (thumbnails.length > 0) {
      thumbnails.forEach((thumbnail) => {
        expect(thumbnail).toBeInTheDocument();
      });
    } else {
      expect(thumbnails.length).toBe(0);
    }
  });
});
