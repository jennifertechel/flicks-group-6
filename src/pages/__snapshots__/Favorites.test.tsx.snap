// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Favorites component > renders correctly with Thumbnail components 1`] = `
<div>
  <div
    class="css-qq6tp8"
  >
    <h2
      class="chakra-heading css-b0qdn7"
    >
      Favorites
    </h2>
    <div
      class="css-j7qwjs"
    >
      <div
        class="css-1khoe0f"
      />
    </div>
  </div>
</div>
`;
